app/models/date_range.rb

```diff
@@ -0,0 +1,30 @@
+class DateRange
+  attr_reader :start_date, :end_date
+  
+  # Step 1: only in use with Conference date range.
+  # Validated in Conference: presence and chronological order
+  #  because the input is done there.
+  
+  def initialize(start_date, end_date)
+    @start_date = start_date
+    @end_date = end_date
+  end
+  
+  # Accepts dates
+  # Returns a string with one date or a formatted range:
+  # 12 - 13 Jul 2017 or 30 Jul - 1 Aug 2017 or
+  # 31 Dec 2017 - 5 Jan 2018 or 1 Dec 2017 - 12 Dec 2018
+  def compact
```

Comment:
The method will raise an error when the input is not a `Date`. So I didn't an `raise Error` at entrance. Should I?